src/catalog/src/information_schema/views.rs

```diff
@@ -0,0 +1,288 @@
+                .map_err(BoxedError::new)
+                .context(InternalSnafu)?,
+        ))
+    }
+}
+
+/// Builds the `information_schema.VIEWS` table row by row
+///
+/// Columns are based on <https://dev.mysql.com/doc/refman/8.4/en/information-schema-views-table.html>
+struct InformationSchemaViewsBuilder {
+    schema: SchemaRef,
+    catalog_name: String,
+    catalog_manager: Weak<dyn CatalogManager>,
+
+    catalog_names: StringVectorBuilder,
+    schema_names: StringVectorBuilder,
+    table_names: StringVectorBuilder,
+    view_definitions: StringVectorBuilder,
+    check_options: StringVectorBuilder,
+    is_updatable: BooleanVectorBuilder,
+    definer: StringVectorBuilder,
+    security_type: StringVectorBuilder,
+    character_set_client: StringVectorBuilder,
+    collation_connection: StringVectorBuilder,
+}
+
+impl InformationSchemaViewsBuilder {
+    fn new(
+        schema: SchemaRef,
+        catalog_name: String,
+        catalog_manager: Weak<dyn CatalogManager>,
+    ) -> Self {
+        Self {
+            schema,
+            catalog_name,
+            catalog_manager,
+            catalog_names: StringVectorBuilder::with_capacity(INIT_CAPACITY),
+            schema_names: StringVectorBuilder::with_capacity(INIT_CAPACITY),
+            table_names: StringVectorBuilder::with_capacity(INIT_CAPACITY),
+            view_definitions: StringVectorBuilder::with_capacity(INIT_CAPACITY),
+            check_options: StringVectorBuilder::with_capacity(INIT_CAPACITY),
+            is_updatable: BooleanVectorBuilder::with_capacity(INIT_CAPACITY),
+            definer: StringVectorBuilder::with_capacity(INIT_CAPACITY),
+            security_type: StringVectorBuilder::with_capacity(INIT_CAPACITY),
+            character_set_client: StringVectorBuilder::with_capacity(INIT_CAPACITY),
+            collation_connection: StringVectorBuilder::with_capacity(INIT_CAPACITY),
+        }
+    }
+
+    /// Construct the `information_schema.views` virtual table
+    async fn make_views(&mut self, request: Option<ScanRequest>) -> Result<RecordBatch> {
+        let catalog_name = self.catalog_name.clone();
+        let catalog_manager = self
+            .catalog_manager
+            .upgrade()
+            .context(UpgradeWeakCatalogManagerRefSnafu)?;
+        let predicates = Predicates::from_scan_request(&request);
+
+        for schema_name in catalog_manager.schema_names(&catalog_name).await? {
+            let mut stream = catalog_manager.tables(&catalog_name, &schema_name);
+
+            while let Some(table) = stream.try_next().await? {
+                let table_info = table.table_info();
+                if table_info.table_type == TableType::View {
+                    let backend_catalog_manager = catalog_manager
+                        .as_any()
+                        .downcast_ref::<KvBackendCatalogManager>()
+                        .context(CastManagerSnafu)?;
+                    let view_info = backend_catalog_manager
```

Comment:
What about extracting the `view_info_cache` out of the loop:
```rust
let view_info_cache = backend_catalog_manager
    .view_info_cache()?;

while let Some(table) = stream.try_next().await? {
      let view_info = view_info_cache.get(table_info.ident.table_id)
                              .await
                              ....
}
```

Reply:
good catch fixed